pkg/operator/resource/resourceapply/apps_test.go

```diff
@@ -134,6 +134,77 @@ func TestApplyDeployment(t *testing.T) {
 	}
 }
 
+func TestDeleteDeployment(t *testing.T) {
+	tests := []struct {
+		name              string
+		desiredDeployment *appsv1.Deployment
+		expectError       bool
+	}{
+		{
+			name:              "when deployment exists",
+			desiredDeployment: workload(),
+			expectError:       false,
+		},
```

Comment:
Perhaps a new test case where the object doesn't exist or can't be deleted?

Reply:
I do not see much value in that test case. Quite frankly, I don't see much value in the test I already wrote but I was just checking the checkboxes. :(

Reply:
Could you please elaborate on why you don't see value in these tests?

I believe it was a reasonable request from @dusk125 because: 1) the functions are new, 2) the tests are straightforward to implement, and 3) they help prevent unintended changes from future refactoring.

We certainly don't have full coverage for everything we add, but IMO it's a good thing to aim for.

Reply:
Sure,  the current test exercises machinery of fake kubeclients. The code in question itself is mostly logic free and if the contract of - `clients.kubeClient.AppsV1()` holds true then it will work as expected. 

IMO - a unit test should test the code it is changing and shouldn't necessarily be testing api clients themselves. So if I add a unit test, which deletes non-existant object or an object that can't be deleted, we are in-truth testing the API client themselves. There is no special handling for that bit in my code.

Reply:
>It doesn't actually return NotFound errors (lines 251, 263), and that's why I wanted to see a test case for deleting a non-existent object. 

Yeah I know. I am dubious about usefulness of a unit test for that, but I am less sure now. :-)

I have added some unit tests for non-existant object deletion fwiw now.